modules/datagrid/72/launch/added/launch/infinispan-config.sh

```diff
@@ -282,17 +282,36 @@ function configure_cache() {
        </encoding>"
   fi
 
-  if [ -n "$(find_env "${prefix}_CACHE_EVICTION_STRATEGY")$(find_env "${prefix}_CACHE_EVICTION_MAX_ENTRIES")" ]; then
-    if [ -n "$(find_env "${prefix}_CACHE_EVICTION_STRATEGY")" ]; then
-      local CACHE_EVICTION_STRATEGY="strategy=\"$(find_env "${prefix}_CACHE_EVICTION_STRATEGY")\""
+  if [ -n "$(find_env "${prefix}_CACHE_EVICTION_MAX_ENTRIES")" ]; then
```

Comment:
I think we need to introduce another variable, such as `_CACHE_MEMORY_EVICTION_SIZE` and deprecate `_CACHE_EVICTION_MAX_ENTRIES` as max entries does not make sense when the eviction type is MEMORY.

If a user specifies the old `..MAX_ENTRIES`, then we issue a warning and just set `_CACHE_MEMORY_EVICTION_SIZE = _CACHE_EVICTION_MAX_ENTRIES` assuming the former hasn't already been set.

Actually, I think we should go a step further and change the variables to `_CACHE_MEMORY_STORAGE_TYPE`, `_CACHE_MEMORY_EVICTION_TYPE`, `_CACHE_MEMORY_EVICTION_STRATEGY` and  `_CACHE_MEMORY_EVICTION_SIZE`. 

We probably need `_CACHE_MEMORY_OFF_HEAP_ADDRESS_COUNT` as well, all though that is very long winded :roll_eyes: